File: src/main/kotlin/de/lotto24/slotmachine/controller/api/ApiController.java
package de.lotto24.slotmachine.controller.api;

import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author skows (initial creation).
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Documented
@RestController
@RequestMapping(path = "api", produces = {MediaType.APPLICATION_JSON_VALUE})
public @interface ApiController {
}

File: src/main/kotlin/de/lotto24/slotmachine/controller/api/ApiController.kt
package de.lotto24.slotmachine.controller.api

import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

/**
 * @author skows (initial creation).
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@RestController
@RequestMapping(path = arrayOf("api"), produces = arrayOf(MediaType.APPLICATION_JSON_VALUE))
annotation class ApiController